src/canvas-effects/canvas-effect.js

```diff
@@ -1794,60 +1602,98 @@ export default class CanvasEffect extends PIXI.Container {
+			file = Sequencer.Database.getEntry(this.data.file).clone();
+		} else {
+			file = SequencerFileBase.make(this.data.file)
+			this._currentFilePath = this.data.file;
+		}
+
+		if (file.template) {
+			this._template = 
+				foundry.utils.mergeObject(
+					{ gridSize: file.template[0], startPoint: file.template[1], endPoint: file.template[2] },
+					this.data.template
+				)
+			
+		}
+		file.fileIndex = this.data.forcedIndex;
+		file.twister = this._twister;
+		this._file = file
+		this._isRangeFind = file?.rangeFind;
+	}
+
+	_updateCurrentFilePath(showDistanceWarning = false) {
+		if (!this._file) {
+			return;
+		}
+		if (!this.data.stretchTo) {
+			this._currentFilePath = this._file.getFile();
+			return;
+		}
+		let ray = new Ray(this.sourcePosition, this.targetPosition);
+		if (ray.distance === 0 && showDistanceWarning) {
+			lib.custom_error(
+			      "effect",
+			      `stretchTo - You are stretching over a distance of "0", you may be attempting to stretch between two of the same coordinates!`
+			);
+		}
+		this._currentFilePath = this._file.getFileForDistance(ray.distance);
+	}
 
 	/**
 	 * Creates the sprite, and the relevant containers that manage the position and offsets of the overall visual look of the sprite
 	 *
 	 * @private
 	 */
-	_createSprite() {
+	async _createSprite() {
 		this.renderable = false;
 
-		const args = [this.spriteSheet ? this.spriteSheet : PIXI.Texture.EMPTY];
-		if (
-			!this.data.xray &&
-			!this.spriteSheet &&
-			!this.data.screenSpace &&
-			!this.data.screenSpaceAboveUI
-		) {
-			args.push(VisionSamplerShader);
-		}
+		// TODO handle / fix vision sampler shader for advanced vision masking?
+		const useVisionMasking = !this.data.xray && !this.data.screenSpace && !this.data.screenSpaceAboveUI;
+		const shader = useVisionMasking ? VisionSamplerShader : undefined
 
-		const spriteType = this.spriteSheet ? PIXI.AnimatedSprite : SpriteMesh;
-		const sprite = new spriteType(...args);
-		this.sprite = this.spriteContainer.addChild(sprite);
+		const spriteData = {
+			antialiasing: this.data?.fileOptions?.antialiasing,
+			tiling: this.data.tilingTexture,
+		}
+		/** @type {SequencerSpriteManager} */
+		this.sprite = new SequencerSpriteManager(this._file, spriteData)
+		this.spriteContainer.addChild(this.sprite)
 		this.sprite.id = this.id + "-sprite";
 
-		Object.values(this._relatedSprites).forEach((sprite) => {
-			if (this.data.tint) {
-				sprite.tint = this.data.tint;
-			}
-			this.sprite.addChild(sprite);
-		});
+		this.sprite.volume = (this.data.volume ?? 0) * game.settings.get("core", "globalInterfaceVolume");
+		this.sprite.loop = this.loops
+		this.sprite.loopDelay = this.loopDelay
+		this.sprite.currentTime = this._startTime
 
-		this.animatedSprite = false;
-		if (this.spriteSheet) {
-			this.animatedSprite = true;
-			this.sprite.animationSpeed = 0.4;
-			this.sprite.loop = false;
+		const activatedSprite = await this.sprite.activate(this._currentFilePath)
```

Comment:
Same here, this variable is unused - is this the equivalent of the stretchTo multi-child sprite system I created?

Reply:
Activated sprite is a remnant of something I tried... I first thought only one sprite would ever be active (which basically means visible/rendered)
I had a system where only the active sprite was modified by the setters of the sprite manager (anchor, Position etc) and then when another sprite was activated the properties of the previously active sprite were copied over. 
Text sprites however kinda messed with the system and the way it is now is more easy to grasp and a cleaner implementation I think. 

In regards to the multi Sprite system. I think so, yes. I first used the activated sprite to access the video textures and so on but that turned out to be cumbersome in some cases so I just added some accessors on the sprite Manager class for that instead.